mpactpy/lattice.py

```diff
@@ -0,0 +1,158 @@
+                            'Y': self.module_map[0][0].pitch['Y']}
+
+        assert all(isclose(self.module_map[i][j].pitch['X'], self.mod_dim['X'])
+                   for j in range(self.ny) for i in range(self.nx))
+
+        assert all(isclose(self.module_map[i][j].pitch['Y'], self.mod_dim['Y'])
+                   for i in range(self.nx) for j in range(self.ny))
+
+        assert all(isclose(module.pitch['Z'], self.module_map[0][0].pitch['Z'])
+                   for row in self.module_map for module in row)
+
+        self._pitch   = {'X': self.mod_dim['X'] * self.nx,
+                         'Y': self.mod_dim['Y'] * self.ny,
+                         'Z': self.module_map[0][0].pitch['Z']}
+
+    def __eq__(self, other: Any) -> bool:
+        if self is other:
+            return True
+        return (isinstance(other, Lattice)     and
+                self.module_map == other.module_map
+               )
+
+    def __hash__(self) -> int:
+        return hash(tuple(tuple(row) for row in self.module_map))
+
+    def write_to_string(self,
+                        prefix: str = "",
+                        module_mpact_ids: Dict[Module, int] = None,
+                        lattice_mpact_ids: Dict[Lattice, int] = None) -> str:
+        """ Method for writing a lattice to a string
+
+        Parameter
+        ---------
+        prefix : str
+            A prefix with which to start each line of the written output string
+        module_mpact_ids : Dict[Module, int]
+            A collection of Modules and their corresponding MPACT IDs
+        lattice_mpact_ids : Dict[Lattice, int]
+            A collection of Lattices and their corresponding MPACT IDs
+
+        Returns
+        -------
+        str
+            The string that represents the lattice
+        """
+
+        if module_mpact_ids is None:
+            modules = unique(module for row in self.module_map for module in row)
+            module_mpact_ids = {module: i+1 for i, module in enumerate(modules)}
+
+        lattice_id = 1 if lattice_mpact_ids is None else lattice_mpact_ids[self]
+
+        id_length = max(len(str(module_mpact_ids[module])) for row in self.module_map for module in row)
+
+        string = prefix + f"lattice {lattice_id} {self.nx} {self.ny}\n"
+        for row in self.module_map:
+            modules = [module_mpact_ids[module] for module in row]
+            string += prefix + prefix + f"{list_to_str(modules, id_length)}\n"
+        return string
+
+    def get_axial_slice(self, start_pos: float, stop_pos: float) -> Lattice:
+        """ Method for creating a new Lattice from an axial slice of this Lattice
+
+        Parameters
+        ----------
+        start_pos : float
+            The starting axial position of the slice
+        stop_pos : float
+            The stopping axial position of the slice
+
+        Returns
+        -------
+        Lattice
+            The new Lattice created from the axial slice
+        """
```

Comment:
Maybe note to the user that if values are out of range, they will be reset to they boundary.  You could also add this note to other classes with `get_axial_slice`. 

```suggestion
        Notes
        -----
        Input positions are automatically clamped to the assembly height range [0, height].
        Positions outside this range will be set to the nearest boundary.
        """
```

Reply:
I added documentation that this will return `None` if values are out of bounds.  I fixed this on the other `get_axial_slice` methods as well. Fixed in #22